packages/@internationalized/date/src/queries.ts

```diff
@@ -233,9 +251,13 @@ function getWeekStart(locale: string): number {
 }
 
 /** Returns the number of weeks in the given month and locale. */
-export function getWeeksInMonth(date: DateValue, locale: string): number {
-  let days = date.calendar.getDaysInMonth(date);
-  return Math.ceil((getDayOfWeek(startOfMonth(date), locale) + days) / 7);
+export function getWeeksInMonth(date: DateValue, locale: string, firstDayOfWeek?: DayOfWeek): number {
+  let monthStart = startOfMonth(date);
+  let firstVisibleDay = startOfWeek(monthStart, locale, firstDayOfWeek);
+  let monthEnd = endOfMonth(date);
+  let lastVisibleDay = endOfWeek(monthEnd, locale, firstDayOfWeek);
+  let totalDays = lastVisibleDay.calendar.toJulianDay(lastVisibleDay) - firstVisibleDay.calendar.toJulianDay(firstVisibleDay) + 1;
+  return Math.ceil(totalDays / 7);
```

Comment:
same logic as before should work if you pass `firstDayOfWeek` through to `getDayOfWeek` I think